### PR TITLE
Add table aws_eks_addon_version. Closes #469

### LINKED ISSUE #469: Add table aws_eks_addon_versions
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

To expose available addons and their EKS control plane version compatibility. 
https://docs.aws.amazon.com/cli/latest/reference/eks/describe-addon-versions.html

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```

SETUP: tests/aws_eks_addon_version []

PRETEST: tests/aws_eks_addon_version

TEST: tests/aws_eks_addon_version
Running terraform
aws_vpc.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creating...
aws_iam_role.named_test_resource: Creation complete after 4s [id=turbottest56922]
aws_iam_role_policy_attachment.named_test_resource: Creating...
aws_iam_role_policy_attachment.named_test_resource2: Creating...
aws_iam_role_policy_attachment.named_test_resource2: Creation complete after 2s [id=turbottest56922-20210607103055322100000001]
aws_iam_role_policy_attachment.named_test_resource: Creation complete after 3s [id=turbottest56922-20210607103055575100000002]
aws_vpc.named_test_resource: Still creating... [10s elapsed]
aws_vpc.named_test_resource: Creation complete after 15s [id=vpc-04464ad728a487c3b]
aws_subnet.named_test_resource2: Creating...
aws_subnet.named_test_resource1: Creating...
aws_subnet.named_test_resource1: Creation complete after 4s [id=subnet-0d144c4e8d833a659]
aws_subnet.named_test_resource2: Creation complete after 4s [id=subnet-0b1f0fb5b041eeeb1]
aws_eks_cluster.named_test_resource: Creating...
aws_eks_cluster.named_test_resource: Still creating... [10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [1m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [2m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [3m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [4m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m30s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m40s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [5m50s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m0s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m10s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m20s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [6m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [7m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [8m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m1s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m11s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m21s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m31s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m41s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [9m51s elapsed]
aws_eks_cluster.named_test_resource: Still creating... [10m1s elapsed]
aws_eks_cluster.named_test_resource: Creation complete after 10m1s [id=turbottest56922]
aws_eks_addon.named_test_resource: Creating...
aws_eks_addon.named_test_resource: Still creating... [10s elapsed]
aws_eks_addon.named_test_resource: Still creating... [20s elapsed]
aws_eks_addon.named_test_resource: Creation complete after 24s [id=turbottest56922:vpc-cni]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "aws eks describe-addon-versions --addon-name vpc-cni > /private/var/folders/d8/gsp6f_cn3359xnt_gs3cts_80000gn/T/tests/aws_eks_addon_version/terraform/test/output.json"]
null_resource.named_test_resource: Creation complete after 4s [id=8457165079498075604]
data.local_file.input: Reading...
data.local_file.input: Read complete after 0s [id=3f34e40766cf314adb5f6d93beb64becc6c25062]
data.template_file.resource_aka: Reading...
data.template_file.resource_aka: Read complete after 0s [id=efcb4030ddfb9480a99b76e2686b634579f77bdad489744747920a78e84cd743]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 9 added, 0 changed, 0 destroyed.

Outputs:

account_id = "013122550996"
addon_version = "v1.7.10-eksbuild.1"
resource_aka = "arn:aws:eks:us-east-1:013122550996:addon-version/v1.7.10-eksbuild.1"

Running SQL query: test-list-query.sql
[
  {
    "account_id": "013122550996",
    "addon_version": "v1.7.10-eksbuild.1"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "013122550996",
    "akas": [
      "arn:aws:eks:us-east-1:013122550996:addon-version/v1.7.10-eksbuild.1"
    ],
    "title": "v1.7.10-eksbuild.1"
  }
]
✔ PASSED

POSTTEST: tests/aws_eks_addon_version

TEARDOWN: tests/aws_eks_addon_version

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  addon_name,
  addon_version,
  type
from
  aws_eks_addon_version;
+------------+--------------------+------------+
| addon_name | addon_version      | type       |
+------------+--------------------+------------+
| vpc-cni    | v1.7.10-eksbuild.1 | networking |
| vpc-cni    | v1.7.6-eksbuild.2  | networking |
| vpc-cni    | v1.7.9-eksbuild.2  | networking |
| vpc-cni    | v1.7.5-eksbuild.2  | networking |
| vpc-cni    | v1.6.3-eksbuild.2  | networking |
| vpc-cni    | v1.7.9-eksbuild.1  | networking |
| kube-proxy | v1.19.6-eksbuild.2 | networking |
| coredns    | v1.8.3-eksbuild.1  | networking |
| vpc-cni    | v1.7.6-eksbuild.1  | networking |
| vpc-cni    | v1.7.5-eksbuild.1  | networking |
| coredns    | v1.8.0-eksbuild.1  | networking |
| vpc-cni    | v1.6.3-eksbuild.1  | networking |
| kube-proxy | v1.18.8-eksbuild.1 | networking |
| coredns    | v1.7.0-eksbuild.1  | networking |
| kube-proxy | v1.20.4-eksbuild.2 | networking |
+------------+--------------------+------------+
> select
  addon_name,
  count(addon_version) as addon_verion_count
from
  aws_eks_addon_version
group by
  addon_name;
+------------+--------------------+
| addon_name | addon_verion_count |
+------------+--------------------+
| coredns    | 3                  |
| kube-proxy | 3                  |
| vpc-cni    | 9                  |
+------------+--------------------+

```
</details>

### CODE REVIEW

File: docs/tables/aws_eks_addon_version.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from
+  aws_eks_addon_version;
+```
+
+### Get addon version count by addon name
+
+```sql
+select
+  addon_name,
+  count(addon_version) as addon_verion_count
```
Comment (Subhajit97) [inline docs/tables/aws_eks_addon_version.md:25]:
```suggestion
  count(addon_version) as addon_version_count
```

File: aws/table_aws_eks_addon_version.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+				Description: "The architectures that the version supports.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "compatibilities",
+				Description: "An object that represents the compatibilities of a version.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe Standard columns
```
Comment (Subhajit97) [inline aws/table_aws_eks_addon_version.go:50]:
```suggestion
			// Steampipe standard columns
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

File: aws/table_aws_eks_addon_version.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	plugin.Logger(ctx).Trace("getAddonVersionAkas")
+	version := h.Item.(addonVersion)
+
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := commonData.(*awsCommonColumnData)
+	akas := []string{"arn:" + commonColumnData.Partition + ":eks:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":addon-version/" + *version.AddonVersion}
```
Comment (cbruno10) [inline aws/table_aws_eks_addon_version.go:119]: @bigdatasourav Is this an ARN format from AWS, or one we made up? If we made it up, would it be better to follow a version ARN format similar to Lambda function versions, e.g., `arn:${Partition}:lambda:${Region}:${Account}:function:${FunctionName}:${Version}`?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comment, thanks!

**Comment by cbruno10** [comment]
@bigdatasourav   How are EKS add-on versions treated in AWS? Are they similar to Lambda function versions, i.e., every time you update a function, a new version is created, and older versions can be accessed/used? Are users able to modify older add-on versions through the console/API, or are they strictly controlled and defined by AWS, and clusters can just select which ones to use?

**Comment by bigdatasourav** [comment]
@cody EKS addon versions are different ,they are managed by aws , we can only select at the time of adding add-on in cluster, Also we can update add-on version to by select the latest available version but we can not modify the old or existing version. Please confirm should we go with the below format like lambda -
arn:${Partition}:eks:${Region}:${Account}:addon:${addonName}:${Version}
or 
arn:${Partition}:eks:${Region}:${Account}:addon:${addonName}:version:${Version}

**Comment by cbruno10** [comment]
@bigdatasourav Let's go with `arn:${Partition}:eks:${Region}:${Account}:addonversion/${AddonName}/${AddonVersion}`, since AWS defines the add-on versions and users can't create/modify/delete them. This format is consistent with the other EKS ARNs.

**Comment by bigdatasourav** [comment]
@cbruno10 ```> select addon_name,addon_version,akas from aws.aws_eks_addon_version
+------------+--------------------+-----------------------------------------------------------------------------------+
| addon_name | addon_version      | akas                                                                              |
+------------+--------------------+-----------------------------------------------------------------------------------+
| vpc-cni    | v1.6.3-eksbuild.2  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.6.3-eksbuild.2"]     |
| kube-proxy | v1.19.6-eksbuild.2 | ["arn:aws:eks:us-east-1:013122550996:addonversion/kube-proxy/v1.19.6-eksbuild.2"] |
| kube-proxy | v1.20.4-eksbuild.2 | ["arn:aws:eks:us-east-1:013122550996:addonversion/kube-proxy/v1.20.4-eksbuild.2"] |
| vpc-cni    | v1.7.9-eksbuild.2  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.9-eksbuild.2"]     |
| kube-proxy | v1.18.8-eksbuild.1 | ["arn:aws:eks:us-east-1:013122550996:addonversion/kube-proxy/v1.18.8-eksbuild.1"] |
| vpc-cni    | v1.7.6-eksbuild.2  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.6-eksbuild.2"]     |
| vpc-cni    | v1.6.3-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.6.3-eksbuild.1"]     |
| vpc-cni    | v1.7.9-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.9-eksbuild.1"]     |
| vpc-cni    | v1.7.10-eksbuild.1 | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.10-eksbuild.1"]    |
| vpc-cni    | v1.7.5-eksbuild.2  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.5-eksbuild.2"]     |
| coredns    | v1.8.3-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/coredns/v1.8.3-eksbuild.1"]     |
| coredns    | v1.8.0-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/coredns/v1.8.0-eksbuild.1"]     |
| vpc-cni    | v1.7.6-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.6-eksbuild.1"]     |
| vpc-cni    | v1.7.5-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/vpc-cni/v1.7.5-eksbuild.1"]     |
| coredns    | v1.7.0-eksbuild.1  | ["arn:aws:eks:us-east-1:013122550996:addonversion/coredns/v1.7.0-eksbuild.1"]     |
+------------+--------------------+-----------------------------------------------------------------------------------+

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**